railgun/internal/diagnostics/server.go

```diff
@@ -2,26 +2,43 @@ package diagnostics
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"net/http/pprof"
 
 	"github.com/go-logr/logr"
-	"github.com/kong/kubernetes-ingress-controller/railgun/internal/manager"
+	"github.com/kong/deck/file"
+	"github.com/kong/kubernetes-ingress-controller/pkg/util"
 )
 
 // Server is an HTTP server running exposing the pprof profiling tool.
 type Server struct {
-	Logger logr.Logger
+	Logger           logr.Logger
+	ProfilingEnabled bool
+	ConfigDumps      util.ConfigDumpDiagnostic
 }
 
+var successfulConfigDump file.Content
```

Comment:
A better way is to re-use a same one variable, did not see any difference when you do the assignment of the variable.
Do I misunderstand ?
```suggestion
var configDump file.Content
```

Reply:
These should remain separate. We do [assign to one or the other](https://github.com/Kong/kubernetes-ingress-controller/blob/8e7c98666d19d6eca2a5b06cc5b1df4c686ca9fb/railgun/internal/diagnostics/server.go#L71-L75) based on whether the received configuration was applied successfully or not.

We do want to keep both failed and successful configuration around and [expose them separately](https://github.com/Kong/kubernetes-ingress-controller/blob/8e7c98666d19d6eca2a5b06cc5b1df4c686ca9fb/railgun/internal/diagnostics/server.go#L103-L104). One of the uses for these dumps is to diff an attempted configuration against the last known good configuration: the rejected items will typically be new items in the diff.

Our logging for individual rejected configuration isn't always clear, and previously-added broken configuration can block the addition of new, valid configuration because we try and apply everything at once. Being able to see the whole diff aides troubleshooting when those complications make the root cause of a failed config unclear.